heron/instance/src/java/com/twitter/heron/instance/bolt/BoltInstance.java

```diff
@@ -96,6 +97,14 @@ public BoltInstance(PhysicalPlanHelper helper,
   }
 
   @Override
+  public void update(PhysicalPlanHelper physicalPlanHelper) {
+    if (bolt instanceof IUpdatable) {
+      ((IUpdatable) bolt).update(physicalPlanHelper.getTopologyContext());
```

Comment:
We invoke user code here which could fail/block. What is the expectation in such cases? I think the `executor` will log the error and launch a new instance if an error occurs?